Src/Particle/AMReX_Particle.H

```diff
@@ -220,6 +220,20 @@ struct Particle
     AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
     ConstParticleIDWrapper id () const & { return ConstParticleIDWrapper(this->m_idcpu); }
 
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    void atomicSetID (const Long id) {
+        uint64_t tmp;
+        ParticleIDWrapper wrapper(tmp);
+        wrapper = id;
```

Comment:
This line can cause a warning with some compilers, since we zero out some of the bits here:
https://github.com/AMReX-Codes/amrex/blob/88b868099018c191e7d3ea424d3e3420b826dd69/Src/Particle/AMReX_Particle.H#L46-L47
but the compiler dues not know that we will overwrite the other half later on as well in `operator=`.

I am not yet sure how to solve this efficiently without having to assign a dummy value to `tmp` that we then immediately overwrite.

```
In file included from /tmp/ci-CqE4ovLK6r/amrex//Src/Particle/AMReX_ParticleTile.H:6,
                 from /tmp/ci-CqE4ovLK6r/amrex//Src/Particle/AMReX_ParticleUtil.H:12,
                 from /tmp/ci-CqE4ovLK6r/amrex//Src/Particle/AMReX_ParticleContainerBase.H:13,
                 from /tmp/ci-CqE4ovLK6r/amrex//Src/Particle/AMReX_Particles.H:5,
                 from ./Source/Particles/WarpXParticleContainer.H:30,
                 from ./Source/Particles/Pusher/GetAndSetPosition.H:11,
                 from ./Source/Particles/Gather/GetExternalFields.H:4,
                 from ./Source/Particles/Gather/FieldGather.H:11,
                 from ./Source/Particles/ElementaryProcess/Ionization.H:11,
                 from ./Source/Particles/PhysicalParticleContainer.H:15,
                 from ./Source/Particles/MultiParticleContainer.H:22,
                 from ./Source/Particles/Collision/BackgroundMCCCollision.H:10,
                 from ./Source/Particles/Collision/CollisionHandler.cpp:9:
/tmp/ci-CqE4ovLK6r/amrex//Src/Particle/AMReX_Particle.H: In member function ‘amrex::Vector<int> ParticleCreationFunc::operator()(const index_type&, ParticleCreationFunc::SoaData_type, ParticleCreationFunc::SoaData_type, ParticleCreationFunc::ParticleTileType**, ParticleCreationFunc::ParticleType*, ParticleCreationFunc::ParticleType*, const ParticleReal&, const ParticleReal&, const index_type*, const amrex::Vector<unsigned int>&, const SmartCopy*, const SmartCopy*, const index_type*, const index_type*, const ParticleReal*) const’:
/tmp/ci-CqE4ovLK6r/amrex//Src/Particle/AMReX_Particle.H:47:17: warning: ‘tmp’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   47 |         m_idata &= 0x00FFFFFF;
      |         ~~~~~~~~^~~~~~~~~~~~~
/tmp/ci-CqE4ovLK6r/amrex//Src/Particle/AMReX_Particle.H:225:18: note: ‘tmp’ was declared here
  225 |         uint64_t tmp;
      |                  ^~~
/tmp/ci-CqE4ovLK6r/amrex//Src/Particle/AMReX_Particle.H:47:17: warning: ‘tmp’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   47 |         m_idata &= 0x00FFFFFF;
      |         ~~~~~~~~^~~~~~~~~~~~~
/tmp/ci-CqE4ovLK6r/amrex//Src/Particle/AMReX_Particle.H:225:18: note: ‘tmp’ was declared here
  225 |         uint64_t tmp;
      |                  ^~~
```

Reply:
I think I'm fine with that solution - could you let me know if https://github.com/AMReX-Codes/amrex/pull/2484 fixes the warning you're seeing?